primitives/application-crypto/src/lib.rs

```diff
@@ -345,22 +347,27 @@ macro_rules! app_crypto_public_common_if_std {
 		}
 
 		impl<'de> $crate::serde::Deserialize<'de> for Public {
-			fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+			fn deserialize<D>(deserializer: D) -> core::result::Result<Self, D::Error>
 			where
 				D: $crate::serde::Deserializer<'de>,
 			{
+				#[cfg(not(feature = "std"))]
+				extern crate alloc;
+				#[cfg(not(feature = "std"))]
+				use alloc::{format, string::String};
```

Comment:
Sorry, my bad. That's regarding fixed point.

Reply:
It now generates the following error:
```
[2023-05-16 08:11:03]   --> bin/node/executor/tests/common.rs:52:3
[2023-05-16 08:11:03]    |
[2023-05-16 08:11:03] 52 |         app_crypto!(sr25519, TEST_KEY_TYPE_ID);
[2023-05-16 08:11:03]    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ use of undeclared crate or module `sp_std`
[2023-05-16 08:11:03]    |
[2023-05-16 08:11:03]    = note: this error originates in the macro `$crate::app_crypto_public_common_if_serde` which comes from the expansion of the macro `app_crypto` (in Nightly builds, run with -Z macro-backtrace for more info)

```
I see the following solutions: Either import `sp_std` in `bin/node/executor`, or switch back to `alloc`. What do you think?

Reply:
`sp_std` in `bin/node/executor` should be fine.

Reply:
I went the middle may: Instead of importing it to node, I introduced a `pub use sp_std::alloc:...` and imported it via `$crate::`. This does not enforce the import of `sp_std` for every crate.
See https://github.com/paritytech/substrate/pull/13027/commits/4850bbacfb5b9e72f8af66b2b3b13cf292643e9e